olp-cpp-sdk-core/src/http/curl/NetworkCurl.cpp

```diff
@@ -348,6 +348,49 @@ void SetupDns(CURL* curl_handle, const std::vector<std::string>& dns_servers) {
 #endif
 }
 
+Diagnostics GetDiagnostics(CURL* handle) {
+  Diagnostics diagnostics;
+
+  static const std::pair<Diagnostics::Timings, CURLINFO> available_timings[] = {
+#if CURL_AT_LEAST_VERSION(8, 6, 0)
+    {Diagnostics::Queue, CURLINFO_QUEUE_TIME_T},
+#endif
+    {Diagnostics::NameLookup, CURLINFO_NAMELOOKUP_TIME_T},
+    {Diagnostics::Connect, CURLINFO_CONNECT_TIME_T},
+    {Diagnostics::SSL_Handshake, CURLINFO_APPCONNECT_TIME_T},
+#if CURL_AT_LEAST_VERSION(8, 10, 0)
+    {Diagnostics::Send, CURLINFO_POSTTRANSFER_TIME_T},
+    {Diagnostics::Wait, CURLINFO_STARTTRANSFER_TIME_T},
+#else
+    {Diagnostics::Send, CURLINFO_STARTTRANSFER_TIME_T},
```

Comment:
haven't dug into the constants description but could be a typo comparing to the line 363